reference/1_application_flow/1_3_synchronization/README.md

```diff
@@ -0,0 +1,140 @@
+
+![](images/2021-07-21-09-16-27.png)
+
+To observe the result, we can wrap this node & drive it with some test data. Note the extra ```flow/Syncer``` here - it takes all 3 arguments produced by the node we just created, to create an output like this:
+
+### Sample output:
+
+```json
+{
+  "param1": "a",
+  "param2": "b",
+  "synced": {
+    "param1": "a",
+    "param2": "b"
+  }
+}
+```
+
+As we can see, the output of the ```flow/Syncer``` is a dictionary of it's input values - as a result, it can also be used for constructing data, as demonstrated in later examples.
+
+## Example - Using parameters with enumeration nodes
+
+> **_Try out:_**
+>
+> Place node **tutorials/app_flow/array_repeater** or **[download .cranqj](cranqj/app_flow_repeater.cranqj)** 
+
+As mentioned in [previously](../1_1_junctions/README.md), iterators & it's derivatives (filters, mappers, reducers) create a new signal with a different tag for each item they process, which cannot be directly synchronized with the original signal. 
+
+The nodes that have this characteristics are the:
+- ```data/array/Iterator```
+- ```data/dictionary/Iterator```
+- ```data/array/Filter```
+- ```data/array/Reducer```
+- ```data/array/Mapper```
+
+Take the case of [**Example - Synchronous filtering**](../1_2_iterators/README.md) as an example - for an array of ```[1, 12, 542]```, the following signals will be propagated:
+
+![](images/2021-08-04-17-16-51.png)
+
+- The ```filter``` node here will create a separate signal with a different tag - represented as separate colours - for each array item
+- It needs this, because it needs to collect the result for every item, before it can summarize & output the result
+- It does so, by making sure, that result for all the tags it sent out are accounted for
+- In Cranq, only signals with matching tags can be synchronized
+
+Now let's implement a node, that takes an array & a numeric value of arguments, adding this value to each element of the array.
+- For this, we can use the ```data/array/Mapper``` node as before
+- Connect an ```number/Add``` to the mapper
+
+If we were to connect the ```value``` input to the ```number/Add``` directly, the following situation would take place: 
+
+![](images/2021-08-04-17-32-18.png)
+
+Our code would not work, because the signals received by the ```number/Add``` from the ```data/array/Mapper``` & and the ```value``` input have different tags, and as such, would not be able to synchronize. Therefore, the adder node would not execute.
+
+To tackle this situation, we must create a signal for the value input as well, with the appropriate tags. To accomplish this, we can use the ```data/array/Repeater``` node:
+
+![](images/2021-08-04-18-01-38.png)
+
+The repeater will perform the same signal forking as the mapper node, creating a new signal carrying our value, with a tag for every array iteration. To allow it to do this, it must also receive the array we are iterating over.

```

Comment:
It seems to me that this example only works because both the mapper and the receiver split signals by adding an sequentially indexed suffix to the tag that previously (here above and also in previous conversations) considered as implementation detail.
I'm unsure whether it is ok to rely on an implementation detail here or we should find an other way to ensure matching signal tags? Ex: store the value in a [Store] and trigger the  <read> input port from the Mapper's <item> output port.

Reply:
We are counting on iteration tags to match up. Perhaps doing this in a string is not ideal - a tag that has structure has been considered before - but for now we can count on this.

Reply:
This is coming up many times - I will add this to the discussion page as well for later discourse